uint32 N
float64[] q
float64[] qd
